src/main/java/org/trigon/hardware/rev/spark/io/SimulationSparkIO.java

```diff
@@ -63,51 +70,30 @@ public void setInverted(boolean inverted) {
     }
 
     @Override
-    public void enableVoltageCompensation(double voltage) {
-        motor.enableVoltageCompensation(voltage);
-    }
-
-    @Override
-    public void setClosedLoopRampRate(double rampRate) {
-        motor.setClosedLoopRampRate(rampRate);
-    }
-
-    @Override
-    public void setSmartCurrentLimit(int limit) {
-        motor.setSmartCurrentLimit(limit);
-    }
-
-    @Override
-    public void setOpenLoopRampRate(double rampRate) {
-        motor.setOpenLoopRampRate(rampRate);
-    }
-
-    @Override
-    public void setPID(double p, double i, double d) {
-        pidController.setP(p);
-        pidController.setI(i);
-        pidController.setD(d);
-    }
-
-    @Override
-    public void setConversionsFactor(double conversionsFactor) {
-        encoder.setConversionsFactor(conversionsFactor);
+    public void setBrake(boolean brake) {
+        SparkMaxConfig configuration = new SparkMaxConfig();
+        configuration.idleMode(brake ? SparkMaxConfig.IdleMode.kBrake : SparkMaxConfig.IdleMode.kCoast);
+        motor.configure(configuration, SparkBase.ResetMode.kNoResetSafeParameters, SparkBase.PersistMode.kNoPersistParameters);
     }
 
     @Override
-    public void restoreFactoryDefaults() {
-        motor.restoreFactoryDefaults();
+    public void updateSimulation() {
+        if (physicsSimulation == null)
+            return;
+        
+        physicsSimulation.setInputVoltage(motorSimulation.getBusVoltage());
+        physicsSimulation.updateMotor();
+        motorSimulation.iterate(physicsSimulation.getSystemVelocityRotationsPerSecond(), RobotHardwareStats.SUPPLY_VOLTAGE, RobotHardwareStats.getPeriodicTimeSeconds());
     }
 
     @Override
-    public void burnFlash() {
-        motor.burnFlash();
+    public void configure(SparkBaseConfig configuration, SparkBase.ResetMode resetMode, SparkBase.PersistMode persistMode) {
+        motor.configure(configuration, resetMode, persistMode);
     }
 
     @Override
-    public void enablePIDWrapping(double minInput, double maxInput) {
-        pidController.setPositionPIDWrappingEnabled(true);
-        pidController.setPositionPIDWrappingMinInput(minInput);
-        pidController.setPositionPIDWrappingMaxInput(maxInput);
+    public void setPhysicsSimulation(MotorPhysicsSimulation physicsSimulation) {
```

Comment:
this should also accept whether it has an absolute or relative encoder
absolute encoder would iterate with system vel, and relative with motor vel